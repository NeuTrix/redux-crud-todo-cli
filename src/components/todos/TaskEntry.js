import React from 'react';
// import PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';

const TaskEntry = props => (
	<TextField
		{...props}
		fullWidth
		label="enter new task"
		margin="dense"
		name="task"
		type="text"
		variant="outlined"
	/>
);

// TaskEntry.propTypes = propTypes;

export default TaskEntry;
