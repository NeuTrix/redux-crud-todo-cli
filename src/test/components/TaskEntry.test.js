/* eslint-env node, mocha */
import React from 'react';
import ReactDOM from 'react-dom';
import { render } from 'enzyme';
import TaskEntry from '../../components/todos/TaskEntry';

it('renders without crashing', () => {
	const div = document.createElement('div');
	ReactDOM.render(
		<TaskEntry
			value="med"
		/>, div,
	);
});

const wrapper = render(
	<TaskEntry
		value="med"
	/>,
);

it('...renders the TaskEntry ', () => {
	expect(wrapper).toMatchSnapshot();
});
