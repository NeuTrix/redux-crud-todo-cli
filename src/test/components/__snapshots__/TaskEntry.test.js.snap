// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`...renders the TaskEntry  1`] = `
<div
  class="MuiFormControl-root-1 MuiFormControl-marginDense-3 MuiFormControl-fullWidth-4"
>
  <label
    class="MuiFormLabel-root-16 MuiFormLabel-filled-20 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13 MuiInputLabel-shrink-12 MuiInputLabel-marginDense-11 MuiInputLabel-outlined-15"
    data-shrink="true"
  >
    enter new task
  </label>
  <div
    class="MuiInputBase-root-36 MuiOutlinedInput-root-23 MuiInputBase-fullWidth-45 MuiInputBase-formControl-37 MuiInputBase-marginDense-43"
  >
    <fieldset
      aria-hidden="true"
      class="MuiPrivateNotchedOutline-root-53 MuiOutlinedInput-notchedOutline-30"
      style="padding-left:8px"
    >
      <legend
        class="MuiPrivateNotchedOutline-legend-54"
        style="width:0"
      >
        <span>
          â€‹
        </span>
      </legend>
    </fieldset>
    <input
      aria-invalid="false"
      class="MuiInputBase-input-46 MuiOutlinedInput-input-31 MuiInputBase-inputMarginDense-47 MuiOutlinedInput-inputMarginDense-32"
      name="task"
      type="text"
      value="med"
    />
  </div>
</div>
`;
